homeassistant/components/frontend/__init__.py

```diff
@@ -769,12 +784,27 @@ def websocket_get_panels(
     hass: HomeAssistant, connection: ActiveConnection, msg: dict[str, Any]
 ) -> None:
     """Handle get panels command."""
+    from .panel_preferences import async_get_panel_preferences  # noqa: PLC0415
+
     user_is_admin = connection.user.is_admin
-    panels = {
-        panel_key: panel.to_response()
-        for panel_key, panel in connection.hass.data[DATA_PANELS].items()
-        if user_is_admin or not panel.require_admin
-    }
+    panel_prefs = async_get_panel_preferences(hass)
+
+    panels = {}
+    for panel_key, panel in connection.hass.data[DATA_PANELS].items():
+        if not user_is_admin and panel.require_admin:
+            continue
+
+        panel_response = panel.to_response()
+
+        # Check if user has set a preference for this panel
```

Comment:
Why are we doing this here in the backend? It conflates frontend navigation data with user preferences?